src/api/app/controllers/webui/package_controller.rb

```diff
@@ -984,18 +984,40 @@ def meta
   end
 
   def save_meta
-    begin
-      frontend.put_file(params[:meta], :project => @project, :package => @package, :filename => '_meta')
-    rescue ActiveXML::Transport::Error => e
-      message = e.summary
-      flash[:error] = message
-      @meta = params[:meta]
-      render :text => message, :status => 400, :content_type => 'text/plain'
+    meta_xml = Xmlhash.parse(params[:meta])
```

Comment:
What happens if the xml is not valid?
Can we use Suse::Validator.validate(schema, content) to ensure that the xml is valid and otherwise show a proper error message (and also test it)!